modules/thermal_hydraulics/src/actions/CoupledHeatTransferAction.C

```diff
@@ -180,9 +198,31 @@ CoupledHeatTransferAction::addUserObjects()
   {
     InputParameters params = class_params;
     params.set<std::vector<VariableName>>("variable") = {_T_solid_var_name};
-    params.set<MooseEnum>("direction") = _direction_enum;
-    params.set<unsigned int>("num_layers") = _num_layers;
     params.set<std::vector<BoundaryName>>("boundary") = {_boundary};
+    if (isParamValid("direction") && isParamValid("num_layers"))
+    {
+      params.set<MooseEnum>("direction") = getParam<MooseEnum>("direction");
+      params.set<unsigned int>("num_layers") = getParam<unsigned int>("num_layers");
+    }
+    else if (isParamValid("position") && isParamValid("orientation") && isParamValid("length") &&
+             isParamValid("n_elems"))
+    {
+      const auto & position = getParam<Point>("position");
+      const auto & orientation = getParam<RealVectorValue>("orientation");
+      const auto & rotation = getParam<Real>("rotation");
+      const auto & lengths = getParam<std::vector<Real>>("length");
+      const auto & n_elems = getParam<std::vector<unsigned int>>("n_elems");
+
+      params.set<MooseEnum>("direction") =
+          DiscreteLineSegmentInterface::getAlignmentAxis(orientation);
```

Comment:
What happens if the orientation is not aligned with one of the axis? I don't see an error message- maybe I missed it.

Reply:
In this case, we get an error from `DiscreteLineSegmentInterface::getElementBoundaryCoordinates`: `Invalid axis.`, but I'll make a check with a better message and then add a test.